app/src/main/java/com/github/k409/fitflow/data/ProfileRepository.kt

```diff
@@ -6,15 +6,16 @@ import javax.inject.Inject
 
 class ProfileRepository @Inject constructor(
     private val db: FirebaseFirestore,
-
+    var success: Boolean = true,
 ) {
     fun submitProfile(
```

Comment:
My suggestion would be to make this method `suspend` and don't use `addOnFailureListener` or `addOnSuccessListener `. In the `suspend` method you could just use `.await()`. It will wait while the operation is complete.

```kotlin
userDocRef.update(updatedData).await()
```